src/FileJobClientBrowser.ts

```diff
@@ -0,0 +1,116 @@
+import axios from "axios";
+
+import { JobClient } from "./JobClient";
+import { Logger } from "./Logger";
+
+import type {
+  ClassInitiator,
+  SubmitJobFileParams,
+  SubmitJobResponse,
+} from "./types";
+
+export class FileJobClient {
+  logger: Logger;
+  readonly baseUrl: string;
+  readonly headers: {
+    Authorization: string;
+  };
+  jobClient: JobClient;
+
+  /**
+   * Creates a JobClient
+   * @param {Object} config object
+   * @param {string} config.url - base url of modzy api (i.e.: https://app.modzy.com/api)
+   * @param {string} config.apiKey - user's API key
+   */
+  constructor({
+    url = "https://app.modzy.com/api",
+    apiKey,
+    logging,
+  }: ClassInitiator) {
+    this.baseUrl = url.endsWith("/") ? url.substring(0, url.length - 1) : url;
+    this.headers = {
+      Authorization: `ApiKey ${apiKey}`,
+    };
+    this.jobClient = new JobClient({ url, apiKey });
+    this.logger = new Logger(logging);
+  }
+
+  uploadFileToOpenJob({
+    file,
+    jobIdentifier,
+    inputName,
+    inputItemKey,
+  }: {
+    file: File;
+    jobIdentifier: string;
+    inputName: string;
+    inputItemKey: string;
+  }): Promise<void> {
+    this.logger.debug("uploadFileToOpenJob", file.name);
+
+    return new Promise(async (resolve, reject) => {
+      const chunkSize = 500 * 1024;
+
+      for (let i = 0; i < file.size; i += chunkSize) {
+        const chunk = file.slice(i, i + chunkSize);
+        const formData = new FormData();
+        formData.append("input", chunk);
+
+        try {
+          const requestUrl = `${this.baseUrl}/jobs/${jobIdentifier}/${inputItemKey}/${inputName}`;
```

Comment:
Per my comment above, lines like this should probably be `${this.baseUrl}/api/jobs/${jobIdentifier}`. If you're comparing with documentation without actually looking at the value of `this.baseUrl` then you'd be confused why `/jobs` when using cURL or something doesn't respond with what you're expecting. It's weird to mix and match domain and path in the same variable.